proc/proc.go

```diff
@@ -101,11 +101,13 @@ func (dbp *Process) Detach(kill bool) (err error) {
 		}
 	}
 	dbp.execPtraceFunc(func() {
-		var sig int
+		err = PtraceDetach(dbp.Pid, 0)
```

Comment:
It doesn't, if kill isn't set the func ends, if it is and err is not nil it returns. But it's convoluted, I've changed it.